src/compiler/utilities.ts

```diff
@@ -3466,6 +3466,32 @@ namespace ts {
     export function getCombinedLocalAndExportSymbolFlags(symbol: Symbol): SymbolFlags {
         return symbol.exportSymbol ? symbol.exportSymbol.flags | symbol.flags : symbol.flags;
     }
+
+    /**
+     * @param writeOnly If set, return true for `x++;` but not for `f(x++);`, which uses `x` in addition to writing to it.
+     */
+    export function isWriteAccess(node: Node, writeOnly: boolean): boolean {

```

Comment:
`isWriteAccess(.., /*writeOnly*/ true)` is ugly allover the place. consider splitting them into two functions, one for `isWriteOnlyAccess`, and one `isReadAccess`